### ISSUE TITLE
FNX-4930 ⁃ Animation: Closing a tab

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#1288]
UX designs: #167 

Closing a tab - Transition duration 120ms

• Animation is reversed (website screen slides down and fades out, tab slides down and fades in).

Animation reference:
https://drive.google.com/file/d/1gg-tCqAKrI08ANVpekGmnltCqvS0o6RS/view?usp=sharing

**Comment by vesta0** [comment]
Moving this out of the sprint.

**Comment by ekager** [comment]
I implemented a similar shared transition that "opens" the tab into the browser, but it does not match the mocks directly. We can revisit this later if it needs tweaking.

**Comment by sblatz** [comment]
@mozilla-mobile/ux can we make sure this animation is considered in the redesign?

**Comment by vesta0** [comment]
This is a product priority since the current experience is jumpy and unpolished.

**Comment by lime124** [comment]
@sblatz are you applying the label as an FYI or is there more information that needs to be provided to implement? If Vesta wants to go ahead and put this in now, we can totally do that and don't need to wait for any redesign efforts.

**Comment by sblatz** [comment]
@lime124 I was just saying that this animation would likely change with the redesign since the tab switcher looks completely different. If @vesta0 thinks it's worth the engineering effort even if we'll rip it out with that redesign, there's no UX feedback needed.

**Comment by lime124** [comment]
thanks for summarizing our disjointed slack conversation 😄

**Comment by vesta0** [comment]
I haven't seen the new tab switcher re-design. If we decide not to work on this ticket because we are waiting for something else, then that issue needs to be on the backlog and sized. I would just rather do something than nothing. If the re-design is on hold, then UX can determine whether the issue reported here can be addressed address in a way that is aligned with the future re-design, or otherwise address it with the originally suggested animation. @lime124 please recommend next steps :) @sblatz please provide sizing for the engineering effort of this ticket as I understand you are concerned about that.  FYI @liuche

**Comment by sblatz** [comment]
I sized the engineering effort, @vesta0 please note that this is still waiting on a google library update that we do not have yet.

**Comment by lime124** [comment]
the effort is currently on hold while we focus on some other musts for the upcoming release. 

i'm going to pass to @topotropic to respond to whether she feels this can be aligned with the future designs. thanks!

**Comment by sblatz** [comment]
@vesta0 is specifically concerned about the "flash" here (when transitioning from browser to home screen). GV is hard to animate and turns black when we navigate away. One solution could be screenshotting the view like we do in focus.

@cpeterso do you know if there's any fix in the works to improve the "black screen flash" when navigating away from a GV engine?

**Comment by cpeterso** [comment]
> @cpeterso do you know if there's any fix in the works to improve the "black screen flash" when navigating away from a GV engine?

Agi on the GV team guesses this problem is because use a `SurfaceView` which is not animatable. He looked into it a while back and it's really not an easy problem to solve. `TextureView` is the alternative which supports all of the animations, but it has a buffer which causes lag (Chrome stopped using it for this reason).

Screenshooting the view and painting it on top of GV content might be one of the few workable (but not ideal) options. Fixing this in the GV compositor would be a lot of work.

**Comment by ekager** [comment]
This seems to be a bit better with the tab animations added back in? At least in the navigation of closing a tab.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/25442310/62733464-8a40f500-b9db-11e9-89da-3f6a54414c0c.gif)

**Comment by sblatz** [comment]
I think this is actually still distracting in light mode. Just dark mode is less noticeable.

**Comment by sblatz** [comment]
![sad 2019-08-09 13_20_19](https://user-images.githubusercontent.com/4400286/62806706-73b49f80-baa8-11e9-9692-07be8d7013a0.gif)
